src/components/timeline/TrackCounterGraph.js

```diff
@@ -25,12 +24,20 @@ import type {
   Thread,
   ThreadIndex,
 } from '../../types/profile';
-import type { AccumulatedCounterSamples } from '../../types/profile-derived';
+import type {
+  AccumulatedCounterSamples,
+  GraphType,
+} from '../../types/profile-derived';
 import type { Milliseconds, CssPixels, StartEndRange } from '../../types/units';
 import type { SizeProps } from '../shared/WithSize';
 import type { ConnectedProps } from '../../utils/connect';
 
-import './TrackMemory.css';
+import './TrackCounter.css';
```

Comment:
I'm torn on this patch, I initially recommended doing a copy/paste of this file, mainly because when a component has bimodal behavior, it gets a lot hard to get it right, and the component is much more complicated. I'd prefer a little code duplication over code complication.

If we keep this as a generic component, then I'd like to see the different behaviors encoded as "feature" checks, not "graph type" checks. I'm going to put my recommendations inline below.